File: rzops-web/src/lib/types/datacenter.ts
export interface DataCenterResponse {
  id: string;
  name: string;
  provider_id: string | null;
  phone: string | null;
  address: string | null;
  country: string | null;
  line_type: string[] | null;
  description: string | null;
  status: string;
  created_at: string;
  updated_at: string;
}

export interface DataCenterListResponse {
  data: DataCenterResponse[];
  count: number;
}

export interface CreateDataCenterRequest {
  name: string;
  provider_id?: string;
  phone?: string;
  address?: string;
  country?: string;
  line_type?: string[];
  description?: string;
  status?: string;
}

export interface UpdateDataCenterRequest extends Partial<CreateDataCenterRequest> {}

export interface ListDataCentersQuery {
  q?: string;
  status?: string;
  country?: string;
  page?: number;
  per_page?: number;
}

File: rzops-api/domain/src/models/data_center.rs
use chrono::{DateTime, Utc};
use uuid::Uuid;


/// DataCenter entity — maps to cmdb_data_center.
#[derive(Debug, Clone)]
pub struct DataCenter {
    pub id: Uuid,
    pub name: String,
    pub provider_id: Option<Uuid>,
    pub phone: Option<String>,
    pub address: Option<String>,
    pub country: Option<String>,
    pub province: Option<String>,
    pub city: Option<String>,
    pub line_type: Option<String>,
    pub description: Option<String>,
    pub status: String,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}
